code/model/DMSDocument.php

```diff
@@ -198,6 +197,18 @@ public function canCreate($member = null)
             }
         }
 
+        // Do early admin check
+        if ($member &&
+            Permission::checkMember(
+                $member,
+                array(
+                    'CMS_ACCESS_DMSDocumentAdmin',
+                )
+            )
+        ) {
```

Comment:
I must say, I'm not a big fan of spreading this code out so much - it's comfortably below the 120 character line limit recommendation without doing this